firefox-ios/Ecosia/Core/Environment/Environment.swift

```diff
@@ -7,17 +7,85 @@ import Foundation
 public enum Environment: Equatable {
     case production
     case staging
+    case debug
 }
 
 extension Environment {
 
     public static var current: Environment {
+        /*
+         * Why not MOZ_CHANNEL_BETA compilation flags?
+         * - Project configs had SWIFT_ACTIVE_COMPILATION_CONDITIONS = ""; blocking xcconfig inheritance
+         * - Multiple BetaDebug configs with same name, Xcode uses wrong one
+         * - EcosiaTesting.xcconfig works because it sets explicit value, not empty string
+         * 
+         * Solution: Bundle ID detection is more reliable than build config inheritance
+         */
+
+        // Determine environment: staging has "firefox" in bundle ID, others use compilation flags
+        guard let bundleId = Bundle.main.bundleIdentifier else { return .production }
+
+        // Staging always has "firefox" in bundle ID (e.g., com.ecosia.ecosiaapp.firefox)
+        if bundleId.contains("firefox") {
+            return .staging
+        }
```

Comment:
Interesting workaround! This feels a bit like hidden magic but I understand the challenges and should be very safe since we don't ever change bundle Ids - so good to go 🙌